backend/dataall/modules/s3_datasets/services/dataset_service.py

```diff
@@ -261,7 +262,7 @@ def update_dataset(uri: str, data: dict):
                     if k not in ['stewards', 'KmsAlias']:
                         setattr(dataset, k, data.get(k))
 
-                DatasetRepository.update_dataset_shares_expiration(session=session, dataset=dataset)
+                DatasetRepository.update_dataset_shares_expiration(session=session, enabledExpiration=dataset.enableExpiration, datasetUri=dataset.datasetUri, expirationDate=ShareObjectService.calculate_expiry_date(expirationPeriod=dataset.expiryMinDuration, expirySetting=dataset.expirySetting))
```

Comment:
Sorry, may be I wasn't 100% clear.
Let's not call one service form another.
Better make `calculate_expiry_date` a private function of DatasetService

Reply:
This would cause repeating the same function in both the ShareObjectService and the DatasetService . Would that be fine ?

Reply:
Also, can you tell me why is it that internal services cannot call each other please ?

Reply:
Updated code. Moved `calculate_expiry_date ` to a `dataall\base\utils` package inside the common_module_utls.py file